tools/plugins/com.liferay.ide.server.ui/src/com/liferay/ide/server/ui/action/RedeployAction.java

```diff
@@ -66,14 +67,21 @@ public void run( IAction action )
 
                 if( liferayServerBehavior != null )
                 {
-                    try
+                    UIUtil.async( new Runnable()
```

Comment:
This will still block the UI once it is called.  We should run this in a background job instead of a UI runnable.